pkg/k8s/client.go

```diff
@@ -100,19 +101,53 @@ func (c *KubeClient) GetPodName(podip string) (podns, poddname string, err error
 	return "", "", fmt.Errorf("match failed, ip:%s matching pods:%v", podip, podList)
 }
 
-func (c *KubeClient) getPodListWithTries(podip string, tries int, sleeptime time.Duration) (*v1.PodList, error) {
-	podList, err := c.PodListWatch.List(metav1.ListOptions{
+func (c *KubeClient) getPodList(podip string) (*v1.PodList, error) {
+	listObject, err := c.PodListWatch.List(metav1.ListOptions{
 		FieldSelector: "status.podIP==" + podip + phaseStatusFilter,
 	})
 
-	if err != nil || len(podList.(*v1.PodList).Items) == 0 {
-		if tries > 1 {
+	if err != nil {
+		return nil, err
+	}
+
+	// Confirm that we are able to cast properly.
+	podList, ok := listObject.(*v1.PodList)
+	if !ok {
+		return nil, fmt.Errorf("list object could not be converted to podllist")
+	}
+
+	if podList == nil {
+		return nil, fmt.Errorf("Podlist nil")
+	}
+
+	if len(podList.Items) == 0 {
+		return nil, fmt.Errorf("Pod List empty")
+	}
+
+	return podList, nil
+}
+
+func (c *KubeClient) getPodListWithTries(podip string, tries int, sleeptime time.Duration) (*v1.PodList, error) {
+	var podList *v1.PodList
+	var err error
+
+	for i := 0; i < tries; i++ {
+		podList, err = c.getPodList(podip)
+		if err != nil {
```

Comment:
can we simplify this to be something like this -

```go
	for i := 0; i < tries; i++ {
		podList, err = c.getPodList(podip)
                if err==nil {
                      break
                }

	        log.Warningf("List pod error: %+v. Retrying, attempt number: %d", err, i)
	        time.Sleep(sleeptime * time.Millisecond)
	}
        return podList, err
```

1. check to see if podList is nil is already done in `getPodList`. So if it's nil we'll get an error.

Reply:
refactored for simplicity and changed the name to reflect the retry functionality.